dora/src/vm/traits.rs

```diff
@@ -46,6 +46,48 @@ pub struct TraitDefinition {
 }
 
 impl TraitDefinition {
+    pub fn new(
+        id: TraitDefinitionId,
+        file_id: FileId,
+        namespace_id: NamespaceId,
+        ast: &Arc<ast::Trait>,
+    ) -> Self {
+        TraitDefinition {
+            id,
+            file_id,
+            namespace_id,
+            is_pub: false,
+            ast: ast.clone(),
+            pos: ast.pos,
+            name: ast.name,
+            type_params: Vec::new(),
+            type_params2: TypeParamDefinition::new(),
+            methods: Vec::new(),
+            instance_names: HashMap::new(),
+            static_names: HashMap::new(),
+            vtables: RwLock::new(HashMap::new()),
+        }
+    }
+
+    pub fn init_modifiers(&mut self, sa: &SemAnalysis) {
+        let annotation_usages = &ast::AnnotationUsages::new();
+        self.is_pub = AnnotationDefinition::is_pub(annotation_usages, sa);
```

Comment:
It seems we always create empty AnnotationUsages and then directly call `is_pub` on it? Am I right and if so, why are we doing this here?

Reply:
I removed the changes that added `AnnotationUsages` to the AST.